src/components/ToggleCollapse.tsx

```diff
@@ -132,34 +126,53 @@ export default function ToggleCollapse(inputProps: PropsWithChildren<ToggleColla
   //// Rendering ////
 
   const toggleId = `toggle-collapse-${props.label.replace(/\s+/g, "-").toLowerCase()}`;
-  const collapseHeight = isExpanded ? contentHeight + "px" : "0";
   // Disable transition on initial rendering so the collapse does not animate
-  const heightTransitionDuration = isInitialRender ? "0ms" : `${ANIMATION_DURATION_MS}ms`;
+  const heightTransitionDuration = isAnimating ? `${ANIMATION_DURATION_MS}ms` : "0ms";
+  const showOverflow = isExpanded && !isAnimating;
+
+  // Choose between a switch and a checkbox for the toggle control
+  let toggleControl = null;
+  if (props.toggleChecked !== undefined) {
+    if (props.toggleType === "checkbox") {
+      toggleControl = (
+        <Checkbox
+          type="checkbox"
+          checked={props.toggleChecked}
+          onChange={(e) => onCheckboxChanged(e.target.checked)}
+          disabled={props.toggleDisabled}
+          id={toggleId}
+        >
+          {props.checkboxLabel}
+        </Checkbox>
+      );
+    } else {
+      toggleControl = (
+        <Switch
+          id={toggleId}
+          checked={props.toggleChecked}
+          onChange={onCheckboxChanged}
+          disabled={props.toggleDisabled}
+          // Align with default label text
+          style={{ paddingTop: "2px" }}
+        />
+      );
+    }
+  }
 
   return (
-    <FlexColumn>
-      <FlexRowAlignCenter style={{ justifyContent: "space-between" }}>
-        <FlexRowAlignCenter $gap={6}>
-          <FlexRowAlignCenter $gap={6}>
-            {props.toggleChecked ? (
-              <label htmlFor={toggleId} style={{ ...defaultProps.labelStyle, ...props.labelStyle }}>
-                {props.label}
-              </label>
-            ) : (
-              <span style={{ ...defaultProps.labelStyle, ...props.labelStyle }}>{props.label}</span>
-            )}
-            {props.toggleChecked !== undefined && (
-              <Switch
-                id={toggleId}
-                checked={props.toggleChecked}
-                onChange={onCheckboxChanged}
-                disabled={props.toggleDisabled}
-                // Align with default label text
-                style={{ paddingTop: "2px" }}
-              />
-            )}
-          </FlexRowAlignCenter>
-          {props.headerContent}
+    <FlexColumn className={"toggle-collapse " + props.className}>
+      <FlexRowAlignCenter className={"toggle-collapse-control-row"} style={{ justifyContent: "space-between" }}>
+        <FlexRowAlignCenter $gap={6} className={"toggle-collapse-header"}>
```

Comment:
Added classnames here for easier external styling (I had to do this for grid alignment in ChannelSettingsControl)